skbio/sequence/_sequence.py

```diff
@@ -547,6 +547,122 @@ def observed_chars(self):
     def _string(self):
         return self._bytes.tostring()
 
+    @classmethod
+    @experimental(as_of="0.4.0-dev")
+    def concat(cls, seqs, how='strict'):
+        """ Concatenate an iterable of Sequence objects.
+
+        Parameters
+        ----------
+        seqs : iterable (Sequence)
+            An iterable of Sequence objects or appropriate subclasses.
```

Comment:
ping on this: describe what an "appropriate subclass" is (what `_assert_can_cast_to` checks for). I think this makes sense in a Notes section.